### PR TITLE
Tests and Updates

### PR DESCRIPTION
**Pull request by koca**
## Tests

Added and updated tests for components

---

### Modal
 - Fix: onClose prop was not working properly for modal and drawers etc
 - Fix: Add removeeventhandler for potential memory leak

#### Why modal or drawer initialFocusRef accepts function instead of pure ref?
when modal mounts, refs which defined inside of modal not exist in `this.$refs`

##### Solutions:
 - initialFocusRef can be a function which returns ref. (Which I did here)
 - Some kind of watch mechanism where it forces to rerun focus traps’ activate function.

---

## Storybook:

Added Drawer component

### CODE REVIEW

**Comment by koca** [comment]
fixes #57

File: packages/chakra-ui-core/src/Drawer/index.js [hunk 1]
```diff
@@ -13,6 +13,10 @@ const Drawer = {
       type: Function,
       default: () => null
     },
+    closeOnEsc: {
```
Comment (codebender828) [inline packages/chakra-ui-core/src/Drawer/index.js]: 👍

File: packages/chakra-ui-core/src/Drawer/tests/Drawer.test.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+// ⚠️ remove skip
+it.skip('should have proper aria', async () => {
+  const inlineAttrs = 'isOpen'
+  const { getByTestId } = renderComponent({ inlineAttrs })
+
+  await Vue.nextTick()
+  const dialog = getByTestId('content') // eslint-disable-line
+
+  // TODO: A11y role dialog or alertdialog?
+  // expect(dialog).toHaveAttribute('role', 'dialog')
```
[the remark this answers is not in the document]
Comment (codebender828) [reply packages/chakra-ui-core/src/Drawer/tests/Drawer.test.js]: But again these I think should be added to another pull request.

**Comment by codebender828** [comment]
hey @koca ! Thanks for working on this. 

I had some changes to suggest to improve the `Modal` component, but I don't think they're related to this PR. I left you a message on Discord about this I'd like to hear you thoughts on ! Cheers!

**Pull request merged**